pkg/operator/ceph/cluster/controller.go

```diff
@@ -259,17 +264,68 @@ func (r *ReconcileCephCluster) reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, cephCluster, errors.Wrap(err, "failed to add finalizer")
 	}
 
+	// After updating the status of the CephCluster, we need to make sure we have the latest
+	refreshCephCluster := func() error {
+		cephCluster, err = r.context.RookClientset.CephV1().CephClusters(cephCluster.Namespace).Get(context.TODO(), cephCluster.Name, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to get latest CephCluster %q changes", request.NamespacedName.String())
+		}
+		return nil
+	}
+
 	// DELETE: the CR was deleted
 	if !cephCluster.GetDeletionTimestamp().IsZero() {
+		// Set the deleting status
```

Comment:
How about we factor out this whole code block into another method for handling the deleted scenario? This reconcile method is getting really long.